measurement/plugins/ip_route/measurements.py

```diff
@@ -0,0 +1,116 @@
+The results are returned in order as:
+ - IPRouteMeasurementResult
+ - LatencyResult of the host used to create the IPRouteMeasurement
+ - The LatencyResults from determining the initial latency
+Traceroute measurement is provided by the "scapy" library. Note that this library needs route privileges to create
+raw sockets, or to have the CAP_NET_RAW capability set.
+
+"""
+import socket
+import validators
+from validators import ValidationFailure
+
+import scapy
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.plugins.ip_route.results import IPRouteMeasurementResult
+from measurement.plugins.latency.measurements import LatencyMeasurement
+
+ROUTE_ERRORS = {
+    "route-err": "iproute encountered an unknown error",
+    "route-address": "iproute failed to find socket address",
+    "route-permission": "iproute does not have permission to create a raw socket",
+}
+
+
+class IPRouteMeasurement(BaseMeasurement):
+    def __init__(self, id, hosts, route_timeout=10, count=4):
+        super(IPRouteMeasurement, self).__init__(id=id)
+
+        if len(hosts) < 1:
+            raise ValueError("At least one url must be provided.")
+        for host in hosts:
+            validated_url = validators.domain(host)
+            if isinstance(validated_url, ValidationFailure):
+                raise ValueError("`{host}` is not a valid host".format(host=host))
+
+        if count < 0:
+            raise ValueError(
+                "A value of {count} was provided for the number of pings. This must be a positive "
+                "integer or `0` to turn off the ping.".format(count=count)
+            )
+
+        self.id = id
+        self.hosts = hosts
+        self.route_timeout = route_timeout
+        self.count = count
+
+    def measure(self):
+        initial_latency_results = self._find_least_latent_host(self.hosts)
+        least_latent_host = initial_latency_results[0][0]
+        results = [self._get_traceroute_result(least_latent_host)]
+        if self.count > 0:
+            latency_measurement = LatencyMeasurement(
+                self.id, least_latent_host, count=self.count
+            )
+            results.append(latency_measurement.measure()[0])
+        results.extend([res for _, res in initial_latency_results])
+        print(results)
+        return results
+
+    def _find_least_latent_host(self, hosts):
+        """
+        Performs a latency test for each specified host
+        Returns a sorted list of LatencyResults, sorted by average latency
+        """
+        initial_latency_results = []
+        for host in hosts:
+            print("host: ", host)
+            latency_measurement = LatencyMeasurement(self.id, host, count=2)
+            initial_latency_results.append((host, latency_measurement.measure()[0]))
+        print("initial latency results: ", initial_latency_results)
```

Comment:
We shouldn't include any `print` statements because the user of the library may not want to see console output.

The right way to do this is with Python logging but there's not yet logging set up in `honestybox-measurements`.